app/src/main/java/org/mascota/ui/view/home/adapter/HomeContentAdapter.kt

```diff
@@ -0,0 +1,32 @@
+package org.mascota.ui.view.home.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import org.mascota.databinding.ItemContentBinding
+import org.mascota.ui.view.home.data.model.HomeContentInfoData
+
+class HomeContentAdapter : RecyclerView.Adapter<HomeContentAdapter.HomeContentViewHolder>(){
+
+    var contentList = listOf<HomeContentInfoData>()
```

Comment:
> 이부분 코드에 대해서 밑에 부분에서 다시 한 번 언급할게요 기억해주세요!

아까 위에서 언급한 리뷰랑 이거랑 동시에 얘기하자면, 저렇게 초기화를 해주고 notify를 해주고 두번의 작업을 해주는 건 살짝 비효율 적인 감이 있기도하고, 캡슐화도 안지켜지구 객체지향적으로도(우리는 delegation 패턴을 씀) 안 맞는다고 생각하기때문에

```suggestion
    private val _contentList = mutableListOf<HomeContentInfoData>()
    var contentList : List<HomeContentInfoData> = _data
        set(value) {
            _data.clear()
            _data.addAll(value)
            notifyDataSetChanged()
        }
```

이렇게 프로퍼티를 활용해서 노티파이랑 초기화랑 같이해주는 게 좋을 것 같아요 :)

Reply:
오 확실히 이게 낫네!!! 그러면 저 set함수는 프래그먼트에서 호출해주면 되는건가요?

Reply:
노놉 프래그먼트에서 쓸때는 그냥 밑에 한 줄만 써도 노티파이랑 초기화랑 같이해줌 ㅎㅎ
```
homeContentAdapter.contentList = it
```

Reply:
왜냐하면 프로퍼티 안에서 setter(그니까 값을 초기화 해줄때 이뤄지는 작업)가 동작하면서 알아서 노티파이도 해줌 ㅎㅎ 그래서 이게 더 깔끔한 방식이라고 생각합니다..

Reply:
아 그렇네! setter 역할이 이래서 좋은거군! 고마워유